tests/experimental/datasources/test_pandas_datasource.py

```diff
@@ -187,17 +213,24 @@ class YearMonth:
         ["-month", "-year"],
     ],
 )
-def test_pandas_sorter(pandas_datasource, csv_path, order_by):
+def test_pandas_sorter(
+    pandas_datasource: PandasDatasource,
+    csv_path: pathlib.Path,
+    order_by: BatchSortersDefinition,
+):
     # Verify test directory has files we expect
     years = ["2018", "2019", "2020"]
     months = [format(m, "02d") for m in range(1, 13)]
-    all_files = os.listdir(csv_path)
+    file_name: PathStr
+    all_files: List[str] = [
+        file_name.stem for file_name in list(pathlib.Path(csv_path).iterdir())
```

Comment:
```suggestion
        file_name.stem for file_name in list(csv_path.iterdir())
```
`csv_path` is already a `Path` so you don't need this